sys/net/gnrc/transport_layer/tcp/gnrc_tcp_option.c

```diff
@@ -11,20 +11,21 @@
  * @{
  *
  * @file
- * @brief       GNRC's TCP option handling related functions
+ * @brief       Implementation of internal/option.h
  *
- * @author      Simon Brummer <brummer.simon@googlemail.com>
+ * @author      Simon Brummer <simon.brummer@posteo.de>
  * @}
  */
 #include "assert.h"
+#include "internal/common.h"
 #include "internal/option.h"
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
 uint32_t _option_build_mss(uint16_t mss)
 {
-    return (((uint32_t )TCP_OPTION_KIND_MSS) << 24) | (((uint32_t) TCP_OPTION_LENGTH_MSS) << 16) | mss;
+    return (uint32_t) ((TCP_OPTION_KIND_MSS << 24) | (TCP_OPTION_LENGTH_MSS << 16) | mss);
```

Comment:
Try the following:

```
return (uint32_t)(((uint8_t)TCP_OPTION_KIND_MSS << 24) | ((uint8_t)TCP_OPTION_LENGTH_MSS) | mss) 
```

I think the cast of the macros to uint8_t is needed otherwise they are int and then you can't shift them by 24.